tests/examples/test_examples.py

```diff
@@ -0,0 +1,80 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Standard
+from os import getenv, path
+import asyncio
+import subprocess
+
+# Third Party
+import pytest
+
+# Local
+from tests.examples.shared import requirements, waitForPort
+
+# E2E tests may be run multiple times in parallel, for that they need an
+# input base port number which is different for every parallel branch
+# Each example test should run on a different increment (100, 200, etc.)
+# to the base grpc port to allow for parallel test execution with no port conflict
+base_grpc_port = int(getenv("GRPC_PORT", "30085"))
+
+
+@pytest.mark.examples
+def test_example_text_sentiment():
+    # Example specific grpc port
+    grpc_port = base_grpc_port + 100
+
+    with requirements("text-sentiment") as (python_venv, example_dir):
+        # Client and server environment variables
+        env_vars = dict(
+            RUNTIME_PORT=str(grpc_port),
+            RUNTIME_METRICS_PORT=str(grpc_port + 1),
+        )
+        # Start the server
+        with subprocess.Popen(
+            [python_venv, "start_runtime.py"],
```

Comment:
If we refactor these two scripts (start_runtime.py and client.py) to have `main()` functions, we could invoke those mains here directly instead of using subprocesses. (The server can be run in a non-blocking mode so the caller doesn't need to create an extra thread or process for it to use)

This is probably not a big deal but I know I am always slightly annoyed when I have to debug something that went wrong in a subprocess vs. being able to spin up a debugger and walk down the full stack